solver-wrapper.js

```diff
@@ -0,0 +1,244 @@
+				eq.lhs = expression;
+				eq.rhs = Parser.parse(0);
+			}
+			if(_isValid(eq)){
+				equations.push(eq);
+				_addVariables(eq);
+			}
+		}
+
+		if(variables.length == 0)
+			throw {
+				type: "no.variables",
+				message: messages.get("no.variables")
+			};
+	}
+
+	/**
+	* used to add unknown variables in the variables array created at the top.
+	* @param -  expression - from which unknown variables are to be added
+	**/
+	var _addVariables = function(/* object */ expression){
+		var update = function(variableArray){
+			for(var index in variableArray){
+				var variable = variableArray[index];
+				if(variables.indexOf(variable) < 0)
+					variables.push(variable);
+			}
+		};
+		update(expression.lhs.variables());
+		if(expression.rhs) update(expression.rhs.variables());
+	};
+
+	/**
+	* Preprocessing of equations will be part of this function. Currently preprocessing
+	* involves removing that equation which does not have any variable in it. 2 = 2 sort
+	* of equations.
+	* @param -  expression - equation which is to checked
+	* @return - isValid - boolean to provide whether expression is correct or not.
+	**/
+	var _isValid = function(/* object */ expression){
+		var isValid = true;
+		if((!expression.lhs || expression.lhs.variables().length == 0) &&
+			(!expression.rhs || expression.rhs.variables().length == 0))
+			isValid = false;
+
+		return isValid;
+	};
+
+	/**
+	* for over determined system where number of equations are more than the number
+	* of variables, this picks up a subset based on removing equations which do not
+	* bring any new variables in the system. First they are labelled as redundant
+	* and then the appropriate equations are removed.
+	* @param -	expressions - array for system of equations to be solved
+	*        	start - this is a search algorithm, index is the starting equation
+	* @return -	excess - the index of equations that were removed from expressions array
+	**/
+	var _preprocess = function(/* Array */ expressions, /* integer */ start){
+		var l = expressions.length;
+		var excess = [];
+		var checked = []; // ids that are part of the system
+		var equationVariables = [];
+		var _comparator = function(a, b){
+			return a - b;
+		};
+		var index = start;
+		
+		for(var i in expressions){
+			isRedundant = true;
+			if(index == l)
+				index = 0;
+			var eq = expressions[index];
+			var ids = lang.clone(eq.lhs.variables());
+			ids = ids.concat(eq.rhs.variables());
+			equationVariables[i] = ids.length;
+
+			for(var j in ids){
+				var id = ids[j];
+				if(checked.indexOf(id) < 0){
+					isRedundant = false;
+					checked.push(id);
+				}
+			}
+
+			if(isRedundant){
+				excess.push(index);
+			}
+			index++;
+		}
+		// by pigeonhole principle this can not happen
+		// but still for safety purposes
+		if(excess.length < l-variables.length)
+			throw{
+				type: "inconsistent.system",
+				message: messages.get("inconsistent.system")
+			};
+
+		if(start === 0 && excess.length > l - variables.length)
```

Comment:
This block, and the two blocks after it could use some comments explaining how they work.

Reply:
Done.